Added herd import for Mellerud

pontus opened the pull request.

pontus committed:
Added herd import for Mellerud

andkaha committed:
Just use a -f test on $3

andkaha committed:
Minor usage info updates and a couple of "cd ... || exit 1" precautions

andkaha committed:
Remove unneeded code, simplify (?) stuff and prettify "Done" message at end

andkaha committed:
Where to put ">&2" redirections (it doesn't matter, as long as it's consistent)

Also, comment out assignment to unused variable.  FIXME comment
added. Should really handle herd registry data files exactly the same
way regardless of whether it's Gotland or Mellerud data.  I will revisit
later.

andkaha committed:
Mostly identical changes as to the other load script

This script will eventually be removed, as soon as I feel confident in
using the Docker loading script.

andkaha commented:
I need to test the actual loading of the Mellerud herd registry data. I haven't had time to do so yet... :-/

jhagberg commented on scripts/load-docker-compose.sh:
```diff
@@ -13,67 +13,9 @@
-if docker-compose -f ../dc-db-load.yml ps | grep -q 'main-load-db.*\sUp\s.*'; then
-    # Alll good here
-    :
+if docker-compose -f ../dc-db-load.yml ps | grep -q 'main-load-db.*\sUp\s.*'
```
This should also check if the herdbook-db container is running or not

andkaha replied:
Will sort that out.

andkaha replied:
> This should also check if the herdbook-db container is running or not

Now fixed with the latest commit (5d9abef5adbf79fb2289efc49018ac853c29f1a0).

pontus commented on scripts/load-docker-compose.sh:
```diff
@@ -82,7 +24,21 @@ else
 
+# Add path "/scripts/" in front of each non-option argument.
+for opt do
+	case $opt in
```
```suggestion
	case "$opt" in
```

andkaha replied:
Quotes actually never needed here.

andkaha replied:
(because `case` is a keyword, not a command, which is why you don't have to quote `$name` in `[[ -f $name ]]` either)

pontus commented on scripts/load-docker-compose.sh:
```diff
@@ -82,7 +24,21 @@ else
 fi
 
+# Add path "/scripts/" in front of each non-option argument.
+for opt do
```
This is obviously ok, but my preference is for explicitness.

andkaha replied:
I wanted to avoid having two _identical_ command line parsing loops.  It's either that I add `/scripts/` in front of the pathnames in the wrapper script, or that I add them later in the actual loading script (`load-in-docker.sh`), in the `while` loop that parses the options.

andkaha committed:
Same logic as in load-mellerud.sh (don't load herd data if file not exist)

andkaha committed:
Make G & M herd registry data be handled the same

Don't avoid loading main data set if herd registry data is not given or
not available.

andkaha commented:
> I need to test the actual loading of the Mellerud herd registry data. I haven't had time to do so yet... :-/

Now tested, and it seems to work well.

Need to fix that thing about making sure that the `herdbook-db` container is up and running.  I'll do that in this PR too since @jhagberg commented on it here.

andkaha committed:
Make sure both herdbook-db and herdbook-main-load-db are up and running

andkaha commented:
I'm ok with this PR a it is now. @pontus or @jhagberg to merge when they wish to do so.

pontus commented:
> I'm ok with this PR a it is now. @pontus or @jhagberg to merge when they wish to do so.

You are stil requested for review, can you approve if you are happy?

andkaha reviewed (approved):
Ok

andkaha commented:
> > I'm ok with this PR a it is now. @pontus or @jhagberg to merge when they wish to do so.
> 
> You are stil requested for review, can you approve if you are happy?

Sorry, done.
